src/commands/TrackerCommand.ts

```diff
@@ -0,0 +1,613 @@
+	if (!activityTrackerBlacklistDb.has(userid)) return [];
+	return activityTrackerBlacklistDb.get(userid);
+}
+
+async function getEntrys(
+	user: string | undefined | null,
+	game: string | undefined | null
+): Promise<string[]> {
+	const allEntrys = activityTrackerLogDb.keyArray();
+	const found: string[] = [];
+	const userCheck: boolean = user == null || user == undefined;
+	const gameCheck: boolean = game == null || game == undefined;
+	allEntrys.forEach((element) => {
+		if (userCheck && gameCheck) {
+			const entry = activityTrackerLogDb.get(element);
+			if (entry !== undefined) found.push(element);
+		} else if (userCheck && !gameCheck) {
+			const gameEntry = element.split("-")[1].toLowerCase();
+			if (gameEntry !== game) return;
+			const entry = activityTrackerLogDb.get(element);
+			if (entry !== undefined) found.push(element);
+		} else if (!userCheck && gameCheck) {
+			const userEntry = element.split("-")[0];
+			if (userEntry !== user) return;
+			const entry = activityTrackerLogDb.get(element);
+			if (entry !== undefined) found.push(element);
+		} else if (!userCheck && !gameCheck) {
+			const split = element.split("-");
+			if (split[0] !== user || split[1].toLowerCase() !== game) return;
+			const entry = activityTrackerLogDb.get(element);
+			if (entry !== undefined) found.push(element);
+		}
+	});
+	return found;
+}
+
+async function makeStats(entrys: string[]): Promise<Array<EmbedField>> {
+	if (entrys.length == 0) {
+		return [];
+	}
+
+	let firstEntry = Infinity;
+	let lastEntry = 0;
+	let playTime = 0;
+	let longestRecord = 0;
+	let totalRecords = 0;
+
+	entrys.forEach((game) => {
+		const logs = activityTrackerLogDb.get(game);
+		logs?.forEach((log) => {
+			totalRecords += 1;
+			playTime += log.t;
+			if (log.w < firstEntry) firstEntry = log.w;
+			if (log.w > lastEntry) lastEntry = log.w;
+			if (log.t > longestRecord) longestRecord = log.t;
+		});
+	});
+
+	let dayDifference = Math.floor(lastEntry / 86400000) - Math.floor(firstEntry / 86400000);
+	dayDifference = dayDifference == 0 ? 1 : dayDifference;
+
+	const average = Math.floor(playTime / dayDifference);
+
+	const fields = [
+		{ name: "Record Range", value: `${dayDifference} Days`, inline: true },
+		{ name: "Total Playtime", value: await makeTimestamp(playTime, true), inline: true },
+		{ name: "Playtime/Day", value: await makeTimestamp(average, false), inline: true },
+		{
+			name: "First Record",
+			value: `<t:${Math.floor(firstEntry / 1000)}> ⁘ <t:${Math.floor(firstEntry / 1000)}:R>`,
+			inline: true,
+		},
+		{
+			name: "Last Record",
+			value: `<t:${Math.floor(lastEntry / 1000)}> ⁘ <t:${Math.floor(lastEntry / 1000)}:R>`,
+			inline: true,
+		},
+		{ name: "Longest Record", value: await makeTimestamp(longestRecord, false), inline: true },
+		{ name: "Total Records", value: `${totalRecords}`, inline: true },
+	];
+
+	return fields;
+}
+
+async function makeTimestamp(ms: number, day: boolean): Promise<string> {
```

Comment:
I already have a util function for that check the reminder util file and move the function in a file in the misc folder and then use it from there

Reply:
please... you could also use the short notation to make this prettier
![image](https://user-images.githubusercontent.com/62220780/198050556-7c3c5d4a-40ff-474d-b9c6-34469a6daf2b.png)